blueprints/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_3_materials.py

```diff
@@ -0,0 +1,132 @@
+        self.f_cm = f_cm
+
+    @staticmethod
+    def _evaluate(
+        beta_cc_t: float,
+        f_cm: MPA,
+    ) -> MPA:
+        """Evaluates the formula, for more information see the __init__ method"""
+        if beta_cc_t < 0:
+            raise ValueError(f"Negative beta_cc_t: {beta_cc_t}. beta_cc_t cannot be negative")
+        if f_cm < 0:
+            raise ValueError(f"Negative f_cm: {f_cm}. f_cm cannot be negative")
+        return beta_cc_t * f_cm
+
+
+class Form3Dot2CoefficientDependentOfConcreteAge(Formula):
+    """Class representing formula 3.2 for the coefficient βcc(t) which is dependent of the age of concrete [-]."""
+
+    label = "3.2"
+    source_document = NEN_EN_1992_1_1_C2_2011
+
+    def __init__(
+        self,
+        s: float,
+        t: int,
+    ) -> None:
+        """Calculates beta_cc(t) coefficient which is dependent of the age of concrete in days [-].
+
+        NEN-EN 1992-1-1+C2:2011 art.3.1.2(6) - Formula (3.2)
+
+        Parameters
+        ----------
+        s: float
+            [s] Coefficient dependent on the kind of cement [-].
+            = 0.20 for cement of strength classes CEM 42.5 R, CEM 52.5 N, and CEM 52.5 R (class R);
+            = 0.25 for cement of strength classes CEM 32.5 R, CEM 42.5 N (class N);
+            = 0.38 for cement of strength class CEM 32.5 N (class S).
+            Use your own implementation of this formula or use the SubForm3Dot2CoefficientTypeOfCementS class.
+        t: int
+            [t] Age of concrete in days [days].
+        """
+        super().__init__()
+        self.s = s
+        self.t = t
+
+    @staticmethod
+    def _evaluate(
+        s: float,
+        t: int,
+    ) -> float:
+        """Evaluates the formula, for more information see the __init__ method"""
+        if s not in (0.20, 0.25, 0.38):
+            raise ValueError(f"Invalid s coefficient: {s}. Options: 0.20, 0.25 or 0.38")
+        if t < 0:
+            raise ValueError(f"Negative t: {t}. t cannot be negative")
+        return np.exp(s * (1 - (28 / t) ** (1 / 2)))
+
+
+class SubForm3Dot2CoefficientTypeOfCementS(Formula):
+    """Class representing sub-formula for formula 3.2, which calculates the coefficient s which is dependent on the cement class"""
+
+    source_document = NEN_EN_1992_1_1_C2_2011
+    label = "3.2"
+
+    def __init__(
+        self,
+        cement_class: str,
+    ) -> None:
+        """[s] Coefficient that depends on the type of cement [-].
+
+        NEN-EN 1992-1-1+C2:2011 art.3.1.2(6) - s
+
+        Parameters
+        ----------
+        cement_class : str
+            [cement_class] Class of the cement.
+                = 'R' for cement of strength classes CEM 42.5 R, CEM 52.5 N, and CEM 52.5 R (class R);
+                = 'N' for cement of strength classes CEM 32.5 R, CEM 42.5 N (class N);
+                = 'S' for cement of strength class CEM 32.5 N (class S).
+
+        """
+        super().__init__()
+        self.cement_class = cement_class
+
```

Comment:
@egarciamendez: are we aware that all the functions we are implementing now don't have latex and intermediate results methods? Or am i missing a step in our roadmap? ;-)

Reply:
Yes, I'm aware and think that we should open a discussion about what the API of those 2 methods should be. At the same time I do think that it doesn't stand in the way of implementing the main logic of the formula right now.

Reply:
@johan-tuls you had a great suggestion the other day about the usage of latex. Should we start from there?
@bro-wi Do we have an (simple) example on the possible interface of the intermediate_results?

Reply:
Agree, just saying so we won't forget ;-)

Reply:
opened a separate issue on this, so we can discuss there

Reply:
> @johan-tuls you had a great suggestion the other day about the usage of latex. Should we start from there? @bro-wi Do we have an (simple) example on the possible interface of the intermediate_results?

I will create a discussion with a code snippet of a method that works really well for what I can see so far